docs/articles/faq/README.md

```diff
@@ -0,0 +1,253 @@
+
+On the server side, you only need to use the `import` statement.
+
+```js
+import fs from 'fs';
+
+fixture `fixture`
+   .page('http://localhost/testcafe/');
+
+test('test', async t => {
+   // var filePath = ...;
+   await t.expect(fs.existsSync(filePath)).ok();
+});
+```
+
+On the client side, use `t.eval` to include the desired module in the test.
+Then you can use this module inside client functions and selectors.
+
+```sh
+test('test', async t => {
+    // eval jquery code to add jQuery to the page
+    await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+
+    var clientFunction = ClientFunction(() => {
+        // You can use $ here
+        return $('div').text();
+    });
+
+    var text = await clientFunction();
+});
+```
+
+### How do I work with configuration files and environment variables?
+
+As TestCafe works with zero configuration, it does not have any config files
+where you can place your custom variables. However, you can always introduce
+your own configuration file and import it to test code.
+
+For example, assume you need to pass a website's base URL to test code.
+In this instance, you may create the following `config.json` file.
+
+```json
+{
+    "baseUrl": "http://localhost/testcafe"
+}
+```
+
+In test code, import it as you would do with a regular JavaScript module.
+
+```js
+import config from './config';
+
+fixture `Fixture`
+    .page `${config.baseUrl}/test1/index.html`;
+```
+
+As an alternative, you may want to use custom command line parameters or environment variables.
+
+The following command passes the `env` argument to the test code.
+
+```sh
+testcafe chrome test.js --env=development
+```
+
+In the test, use an argument parser library (like `minimist`) to parse custom arguments.
+
+```js
+import minimist from 'minimist';
+
+const args = minimist(process.argv.slice(2));
+const environment = args.env;
+
+fixture('example')
+    .page('http://example.com');
+
+test('check environment', async t => {
+  console.log('Environment:', environment);
+});
+```
+
+To set an environment variable use the following command on Windows.
+
+```sh
+set DEV_MODE=true
+testcafe chrome test.js
+```
+
+In test code, you can access this variable as `process.env.DEV_MODE`.
+
+## Troubleshooting
+
+### I have installed TestCafe but I cannot run it. What's your advice?
+
+First, make sure that your firewall does not block ports used by TestCafe.
+By default, TestCafe chooses free ports automatically. That is why it can be
+hard to determine which ports it actually uses. In this instance, use
+the [--ports](../documentation/using-testcafe/command-line-interface.md#--ports-port1port2) command line option
+or the [createTestCafe](../documentation/using-testcafe/programming-interface/createtestcafe.md)
+API factory function to specify custom ports. After that, you can check that these specific ports are not blocked by the firewall.
+
```

Comment:
**Check your firewall.** First, make sure that your firewall does not block ports used by TestCafe. By default, TestCafe chooses free ports automatically. Use the [--ports](http://strelyaev-w8:8080/testcafe/documentation/using-testcafe/command-line-interface.html#--ports-port1port2) command line option or the [createTestCafe](http://strelyaev-w8:8080/testcafe/documentation/using-testcafe/programming-interface/createtestcafe.html) API factory function to specify custom ports. After that, check that these specific ports are not blocked by the firewall.